Multivariant system test support for TSDB enabled package (#8827)
elastic/integrations: docs/developer_tsdb_migration_guidelines.md

# <a id="troubleshooting"></a> Troubleshooting

> ### Enabling multiple variants support in system tests

Reviewer: Is the Kibana version requirement only for TSDB-enabled packages, or does it apply to any package with multiple variants? If it's specific to TSDB-enabled packages, we can change it to something like: "Configuring system tests for TSDB-enabled packages."

Author: It applies only when you have multi-variants in system tests for a TSDB-enabled package. For other packages, this criteria does not apply.

Reviewer: I think then we can update this to something like:
```suggestion
### Configuring system tests for TSDB-enabled packages
```

Reviewer: let's leave it as is since it's TSDB-specific documentation.

> If your system test configuration for a TSDB enabled package contains multiple variants, make sure to set the `kibana.version` in the `manifest.yml` file of the package to be `8.8.1` or higher.

Reviewer: ```suggestion
If the system test configuration for a TSDB-enabled package contains multiple variants, make sure to set the `kibana.version` specified in the `manifest.yml` file of the package to `8.8.1` or a version higher.
```

> If the system test configuration for a TSDB-enabled package contains multiple variants, make sure to set the `kibana.version` specified in the `manifest.yml` file of the package to `8.8.1` or a version higher.

Reviewer 2: This is actually an issue with any attempt to run the same system tests multiple times in the same deployment. I wonder if we should investigate some way to address this case in `elastic-package`.

I think the issue was fixed in https://github.com/elastic/elasticsearch/pull/96461.
